pset5/speller/dictionary.c

```diff
@@ -112,72 +138,70 @@ bool load(const char *dictionary)
         return false;
     }
     
-    // For each word in the dictionary, hash it into the
-    // hash table
     char word[LENGTH + 1] = {0};
-    printf("size of word: %lu\n", sizeof(word));
+    
+    // initialze root, do I have to initiate root TODO
+    root = calloc(1, sizeof(node));
+    // initialize wordCount
+    wordCount = 0;
+    
     while (fscanf(dic_file, "%s", word) != EOF) {
-        
-        // make new node for word
-        node *new_node = malloc(sizeof(node));
-        // Abort if we run out of space
-        if (new_node == NULL) {
-            unload();
-            return false;
-        }
+        wordCount++;
+        insert_word(root, word);
 
-        // Initialize next pointer
-        new_node -> next = NULL;
-        // Copy word into node
-        strcpy(new_node -> word, word);
-        
-        // get word's hash number
-        int index = adler32(new_node -> word, sizeof(new_node -> word));
-        // If there isn't a linked list in that index's bucket yet
-        // insert the node to be the head of that particular linked list
-        if (hashtable[index] == NULL) {
-            hashtable[index] = new_node;
-        }
-        else {
-            // Add the new_node to the linked list that is already there
-            // tne new_node will become the new HEAD
-            node *old_head = hashtable[index];
-            new_node -> next = old_head;
-            hashtable[index] = new_node;
-        }
-        
     }
+
     // close dictionary file
     fclose(dic_file);
+    loaded = true;
     return true;
 }
 
+/** Returns number of words found in *node n
+unsigned int find_words(node *n) {
+    int i;
+    for (i = 0; i < CHAR_AMOUNT; i++) {
+        if (n -> )
+    }
+}*/
+
 /**
  * Returns number of words in dictionary if loaded else 0 if not yet loaded.
  */
 unsigned int size(void)
 {
-    // Check if the hashtable has anything in it
+    // Check if the trie has anything in it
+    if (loaded) {
```

Comment:
for short and simple things like this, the ternary operator is common.

`return loaded ? wordcount :  0;`